### DISCUSSION TITLE
Suggestions for RSDS development

### DISCUSSION THREAD

**Discussion by tautvilas** [pundid/did-instance-nodejs#1] [Ideas]
What are your ideas how RSDS should operate?

**Comment by toomim** [comment]
There needs to be a protocol spec. No system is distributed unless its nodes can be implemented by others, which requires a protocol spec!

Then we can compare and contrast the design decisions with activitypub, at proto, rss, matrix, etc directly.

I am working on a similar system. I need a way to compare my design with yours, and see if we can make them interoperate!

**Reply by tautvilas** [reply]
Hi, thanks for the comment, maybe you also have a link to your system documentation for reference?

**Reply by toomim** [reply]
Yes, I'm building it as a data schema on [Braid-HTTP](https://braid.org).
- Spec: https://braid.org/apps/braidmail/spec2
- Demo: https://mail.braid.org

Braid extends HTTP from a state *transfer* into a generic state *synchronization* protocol. Then we simply define a schema for users and posts, and braid facilitates synchronizing that data across servers and clients, creating a lovely decentralized social system.

The result is a lot like RSS, but more articulate, and provides realtime updates, deletes, and edits of state. Hosting content is super simple—you just need to host JSON blobs somewhere. You can use github pages, or JSBin. And if you add Braid features, you'll get realtime updates. You can also add your own social features (such as likes, follower lists, moderation) by adding new data with new schemas. The other features you [want](http://tautvilas.lt/decentralized-syndication-the-missing-internet-protocol/) are all within scope. You seem to basically want a more advanced HTTP, and that's what we're building!
